tubular/mapping.py

```diff
@@ -150,97 +173,69 @@ class MappingTransformer(BaseMappingTransformer, BaseMappingTransformMixin):
+    for each columns, this dict will return the key i.e. the original value in that row if it
+    is not available in the mapping dict.
 
-                warnings.warn(
-                    f"{self.classname()}: No values from mapping for {col} exist in dataframe."
-                )
+    This transformer inherits from BaseMappingTransformMixin as well as the BaseMappingTransformer
+    in order to access the standard pd.Series.map transform function.
 
-            if len(values_to_be_mapped.difference(values_in_df)) > 0:
+    Parameters
+    ----------
+    X : pd.DataFrame
+        Data with nominal columns to transform.
 
-                warnings.warn(
-                    f"{self.classname()}: There are values in the mapping for {col} that are not present in the dataframe"
-                )
+    suppress_dtype_warning: Bool, default = False
+        Whether to suppress warnings about dtype changes
 
-        X = BaseMappingTransformMixin.transform(self, X)
+    Returns
+    -------
+    X : pd.DataFrame
+        Transformed input X with levels mapped accoriding to mappings dict.
 
-        mapped_dtypes = X[mapped_columns].dtypes
+    """
+    mapped_columns = self.mappings.keys()
 
-        if not suppress_dtype_warning:
+    # Store the original dtypes of the columns
+    original_dtypes = X[mapped_columns].dtypes
 
-            for col in mapped_columns:
+    for col in mapped_columns:
+        values_to_be_mapped = set(self.mappings[col].keys())
+        values_in_df = set(X[col].unique())
 
-                col_mappings = pd.Series(self.mappings[col])
-                mapping_dtype = col_mappings.dtype
+        if len(values_to_be_mapped.intersection(values_in_df)) == 0:
+            warnings.warn(
+                f"{self.classname()}: No values from mapping for {col} exist in dataframe."
+            )
 
-                if (mapped_dtypes[col] != mapping_dtype) and (
-                    mapped_dtypes[col] != original_dtypes[col]
-                ):
+        if len(values_to_be_mapped.difference(values_in_df)) > 0:
+            warnings.warn(
+                f"{self.classname()}: There are values in the mapping for {col} that are not present in the dataframe"
+            )
 
-                    # Confirm the initial and end dtypes are not categories
-                    if not (
-                        is_categorical_dtype(original_dtypes[col])
-                        and is_categorical_dtype(mapped_dtypes[col])
-                    ):
+    X = BaseMappingTransformMixin.transform(self, X)
 
-                        warnings.warn(
-                            f"{self.classname()}: This mapping changes {col} dtype from {original_dtypes[col]} to {mapped_dtypes[col]}. This is often caused by having multiple dtypes in one column, or by not mapping all values."
-                        )
+    # Call the check_dtype_changes method
+    self.check_dtype_changes(X, original_dtypes, suppress_dtype_warning)
```

Comment:
why are we doing the dtype changes check in two places? both in `BaseMappingTransformMixin.transform` and after it?